pkg/webhook/resources/upgrade/validator.go

```diff
@@ -26,9 +30,11 @@ import (
 )
 
 const (
-	upgradeStateLabel     = "harvesterhci.io/upgradeState"
-	skipWebhookAnnotation = "harvesterhci.io/skipWebhook"
-	managedChartNamespace = "fleet-local"
+	upgradeStateLabel       = "harvesterhci.io/upgradeState"
+	skipWebhookAnnotation   = "harvesterhci.io/skipWebhook"
+	rkeInternalIPAnnotation = "rke2.io/internal-ip"
+	managedChartNamespace   = "fleet-local"
+	minFreeDiskSpace        = 30 * 1024 * 1024 * 1024 // 30GB
```

Comment:
`30 GB` is an estimated value, it may not always be fit in different releases, or for all user environments, e.g. from v1.2.0 to v1.2.1 it needs min 35 GB.

more flexible solution will be:

let it be a Harvester setting, or read certain `label/annotation` from `Upgrade` to overwrite this value

the Harvester controller/webhook POD will not be upgraded before an upgrade, those hardcoded value may be a barrier, it needs a way to load those requirements dynamically.

cc @bk201

Reply:
Yes, it's not good to hard code here. I think it's good to add an annotation like `harvesterhci.io/minFreeDiskSpace` when we release [version.yaml](https://releases.rancher.com/harvester/v1.1.1/version.yaml). If we can't find the annotation, we use the default value (30 GB).